extensions/cassandra-client/deployment/src/main/java/io/quarkus/cassandra/deployment/CassandraClientProcessor.java

```diff
@@ -0,0 +1,115 @@
+package io.quarkus.cassandra.deployment;
+
+import static io.quarkus.deployment.annotations.ExecutionTime.RUNTIME_INIT;
+import static io.quarkus.deployment.annotations.ExecutionTime.STATIC_INIT;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.Produces;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+
+import io.quarkus.arc.Unremovable;
+import io.quarkus.arc.deployment.BeanContainerListenerBuildItem;
+import io.quarkus.arc.deployment.GeneratedBeanBuildItem;
+import io.quarkus.arc.deployment.GeneratedBeanGizmoAdaptor;
+import io.quarkus.cassandra.config.CassandraClientConfig;
+import io.quarkus.cassandra.runtime.AbstractCassandraClientProducer;
+import io.quarkus.cassandra.runtime.CassandraClientRecorder;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.ConfigurationBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.recording.RecorderContext;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.ClassOutput;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+import io.quarkus.smallrye.health.deployment.spi.HealthBuildItem;
+
+class CassandraClientProcessor {
+
+    @SuppressWarnings("unchecked")
+    @Record(STATIC_INIT)
+    @BuildStep
+    BeanContainerListenerBuildItem build(
+            RecorderContext recorderContext,
+            CassandraClientRecorder recorder,
+            BuildProducer<FeatureBuildItem> feature,
+            BuildProducer<GeneratedBeanBuildItem> generatedBean) {
+
+        feature.produce(new FeatureBuildItem(FeatureBuildItem.CASSANDRA_CLIENT));
+
+        String cassandraClientProducerClassName = getCassandraClientProducerClassName();
+        createCassandraClientProducerBean(generatedBean, cassandraClientProducerClassName);
+
+        return new BeanContainerListenerBuildItem(recorder.addCassandraClient(
+                (Class<? extends AbstractCassandraClientProducer>) recorderContext
+                        .classProxy(cassandraClientProducerClassName)));
+    }
+
+    private String getCassandraClientProducerClassName() {
+        return AbstractCassandraClientProducer.class.getPackage().getName() + "."
+                + "CassandraClientProducer";
+    }
+
+    private void createCassandraClientProducerBean(
+            BuildProducer<GeneratedBeanBuildItem> generatedBean,
+            String cassandraClientProducerClassName) {
```

Comment:
I was following the pattern used for `mongo-client` in `MongoClientProcessor` and they are also using auto-generation. Right now it is the simplest `buildItem` but probably it will evolve and we would need to do the generation of this code so I propose to keep it this way right now (at least until all needed tasks will be done) then we will see if this code can be simplified, ok?

Reply:
How about going the other way: start simple, and add complexity only when we need it.

I spent a lot of time figuring out this method, it's even harder to understand if it does something that's not strictly necessary.

Reply:
I've tried to simplify this as you suggested.
Unfortunately, there are some problems with order of initialization, and this:
```
 public RuntimeValue<CqlSession> getClient() {
    return new RuntimeValue<>(Arc.container().instance(CqlSession.class, defaultName()).get());
  }
```
is failing with null pointer, because there is no `CqlSession` yet, I'll try to investigate more